app/controllers/concerns/effective_user.rb

```diff
@@ -3,11 +3,11 @@ def effective_user_id
     [
       session[:ial2_recovery_user_id],
       session[:doc_capture_user_id],
-      current_user&.id,
     ].find(&:present?)
   end
 
   def effective_user
-    User.find_by(id: effective_user_id) if effective_user_id
+    return User.find_by(id: effective_user_id) if effective_user_id
```

Comment:
I see we're not using it directly anymore, but if `effective_user_id` is a public method, I'd expect it to return the user ID corresponding to `effective_user` (the test case that was removed).

Maybe:

- Make `effective_user_id` private
- Or, keep `effective_user_id` as it was previously, and do some logic to check if `effective_user_id == current_user.id` to shortcut the query ?

Reply:
I like both!

`effective_user_id` isn't used publicly except to be tested, so I didn't change the implementation and moved it to `private`. The `effective_user` call now will return `current_user` if `effective_user_id` returns the same id on `current_user`.

Reply:
Works for me!